IPA-End-to-End.md

```diff
@@ -231,7 +231,9 @@ To generate the encrypted _match key_, the _user agent_ will perform the followi
 
 The key privacy property in generating the _encrypted match key_ is to prevent the _report collector_ from learning anything new about the user from the _user agent_. As a different secret sharing and encryption is used across the contexts of different _websites/apps_, _encrypted match keys_ cannot be used by the _report collector_ to enable cross-context tracking. Since an encrypted _match key_ is always returned upon request, the existence of a report reveals no new information to the _report collector_. Finally, since a random fallback value is used, a _report collector_ cannot learn even if the _match key_ was set.
 
-But there is another key privacy property enabled by _match keys_, and that is capping the total contribution a single user can make to the aggregate output of an IPA query. As such, _report collectors_ will need to make a commitment (for at least an _epoch_) to a specific _match key provider_, to prevent exceeding the intended privacy budget by leveraging multiple providers. What’s more, _match key_ providers will not be able to update the value of the _match key_ stored in a user-agent until the start of the next _epoch_.
+An _encrypted match key_ is bound both to the _site_ that requested it and to the _epoch_ in which it was generated.  This prevents a site from misrepresenting where the _match key_ was generated and having it used for multiple queries.
+
+But there is another key privacy property enabled by _match keys_, and that is capping the total contribution a single user can make to the aggregate output of an IPA query. As such, each _site_ (or the _report collector_ they delegate to) will need to make a commitment (for at least an _epoch_) to a specific _match key provider_, to prevent exceeding the intended privacy budget by leveraging multiple providers. What’s more, _match key_ providers will not be able to update the value of the _match key_ stored in a user-agent until the start of the next _epoch_.
```

Comment:
```suggestion
But there is another key privacy property enabled by _match keys_, and that is capping the total contribution a single user can make to the aggregate output of an IPA query. As such, each _site_ (or the _report collector_ they delegate to) will need to make a commitment (for at least an _epoch_) to a specific _match key provider_, to prevent exceeding the intended privacy budget by leveraging multiple providers. What’s more, although a _match key_ provider can call `set_matchkey` at any time, any change they make will not be applied by the user agent until the start of the next _epoch_.
```

Reply:
I think that I prefer that over yours because yours implies that the delay is conditional on there being a match key set already.  It should be unconditional.

Reply:
hmm. it should certainly be oblivious to the match key provider and anyone getting an encrypted match key, but if no match key exists, shouldn't it take effect immediately upon the first call to `setMatchKey()`?

Reply:
I don't think so.  The "attack" we're concerned about is increasing the number of match keys that might be attributed to a person within an epoch.  If we move from a backup match key to a real one, that's the same as moving from one real match key to another.  So we have to wait either way.

Reply:
Yea, ok, I follow. This seems potentially problematic (from a utility perspective,) and worth figuring out if we can support it without enabling this "attack", but I'll move that to a new issue.